### PR TITLE
add in burned event

### PR DESCRIPTION
**Pull request by AnonJon**
- rename `ccip_tokens_locked` to `ccip_tokens_transferred`
- include `burned` in events in `ccip_tokens_transferred`
- include `tx_hash` in `ccip_send_requested`

### CODE REVIEW

File: models/chainlink/arbitrum/chainlink_arbitrum_ccip_tokens_transferred_logs.sql [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  block_number,
+  block_time,
+  index,
+  tx_index,
+  tx_from,
+  bytearray_to_uint256(bytearray_substring(data, 1, 32)) as total_tokens
+FROM
+  {{ source('arbitrum', 'logs') }} logs
+WHERE
+  topic0 = 0x9f1ec8c880f76798e7b793325d625e9b60e4082a553c98f42b6cda368dd60008 -- Locked
```
Comment (Hosuke) [inline models/chainlink/arbitrum/chainlink_arbitrum_ccip_tokens_transferred_logs.sql:28]: Do you want to add incremental filters for selection from `logs` tables, as they are large tables.

File: models/chainlink/arbitrum/chainlink_arbitrum_ccip_tokens_transferred_logs.sql [hunk 2]
```diff
@@ -0,0 +1,49 @@
+{{
+  config(
+    
+    alias='ccip_tokens_transferred_logs',
+    materialized='view'
```
Comment (Hosuke) [inline models/chainlink/arbitrum/chainlink_arbitrum_ccip_tokens_transferred_logs.sql:5]: We can add schema in the config blocks and change the materialized strategy to `incremental`.

**Review by Hosuke — CHANGES_REQUESTED** [review]

**Comment by AnonJon** [comment]
@Hosuke I added in the filter. Let me know if that looks right.

File: models/chainlink/avalanche_c/chainlink_avalanche_c_ccip_tokens_transferred_logs.sql [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  tx_index,
+  tx_from,
+  bytearray_to_uint256(bytearray_substring(data, 1, 32)) as total_tokens
+FROM
+  {{ source('avalanche_c', 'logs') }} logs
+WHERE
+  topic0 = 0x9f1ec8c880f76798e7b793325d625e9b60e4082a553c98f42b6cda368dd60008 -- Locked
+  AND block_time >= CAST('2023-07-06' AS date)
+  {% if is_incremental() %}
+            AND block_time >= date_trunc('day', now() - interval '{{incremental_interval}}' day)
```
Comment (Hosuke) [inline models/chainlink/avalanche_c/chainlink_avalanche_c_ccip_tokens_transferred_logs.sql:34]: We can use the `incremental_predicate` here:
```suggestion
            AND {{ incremental_predicate('block_time') }}
```

**Comment by AnonJon** [comment]
@Hosuke updated now with incremental_predicate

File: models/chainlink/polygon/chainlink_polygon_ccip_tokens_transferred_logs.sql [hunk 4]
```diff
@@ -0,0 +1,59 @@
+{{
+  config(
+    alias='ccip_tokens_transferred_logs',
```
Comment (Hosuke) [inline models/chainlink/polygon/chainlink_polygon_ccip_tokens_transferred_logs.sql:3]: You can add the schema in all config blocks now:
```suggestion
  config(
    schema='chainlink_polygon',
    alias='ccip_tokens_transferred_logs',
```

File: models/chainlink/polygon/chainlink_polygon_ccip_tokens_transferred_logs.sql [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  block_time,
+  index,
+  tx_index,
+  tx_from,
+  bytearray_to_uint256(bytearray_substring(data, 1, 32)) as total_tokens
+FROM
+  {{ source('polygon', 'logs') }} logs
+WHERE
+  topic0 = 0x9f1ec8c880f76798e7b793325d625e9b60e4082a553c98f42b6cda368dd60008 -- Locked
+  AND block_time >= CAST('2023-07-06' AS date)
```
Comment (Hosuke) [inline models/chainlink/polygon/chainlink_polygon_ccip_tokens_transferred_logs.sql:31]: We can introduce jinja var like `project_start_date` for the dates used in the sql:
i.e.
`{% set project_start_date = '2023-07-06' %}`
and
`AND block_time >= TIMESTAMP '{{project_start_date}}'`

File: models/chainlink/polygon/chainlink_polygon_ccip_tokens_transferred_logs.sql [hunk 6]
```diff
@@ -0,0 +1,59 @@
+{{
+  config(
+    alias='ccip_tokens_transferred_logs',
+    materialized='incremental',
+    file_format='delta',
+    incremental_strategy='merge',
+  )
```
Comment (Hosuke) [inline models/chainlink/polygon/chainlink_polygon_ccip_tokens_transferred_logs.sql:7]: You need to add the `unique_key` in the config blocks for incremental updates:
```suggestion
    incremental_strategy='merge',
    unique_key=['tx_hash', ...]
  )
```

File: models/chainlink/polygon/chainlink_polygon_schema.yml [hunk 7]
```diff
@@ -1396,15 +1397,15 @@ models:
...
     meta:
       blockchain: "polygon"
       project: chainlink
       contributors: ["linkpool_jon"]
     config:
-      tags: ["chainlink", "ccip", "tokens", "locked", "logs", "polygon"]
+      tags: ["chainlink", "ccip", "tokens", "transferred", "logs", "polygon"]
     description: >
-      Chainlink CCIP Tokens Locked Logs
+      Chainlink CCIP Tokens Transferred Logs
```
Comment (Hosuke) [inline models/chainlink/polygon/chainlink_polygon_schema.yml:1408]: Remember to add tests for `unique_key`
```
tests:
      - dbt_utils.unique_combination_of_columns:
          - ...
```

**Review by Hosuke — CHANGES_REQUESTED** [review]
Some adjustments needed for incremental views.

**Comment by AnonJon** [comment]
@Hosuke added in the unique tests

**Review by Hosuke — APPROVED** [review]
LGTM.✅
Thank you @AnonJon 

[Sample chainlink_ccip_tokens_transferred_logs](https://dune.com/queries/3660861)

Also please run `git merge main` to update this PR.

**Comment by Hosuke** [comment]
Hi @AnonJon, please run `git merge main` to update this PR as well.

**Review by jeff-dude — APPROVED** [review]

**Pull request merged**